prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -240,11 +248,19 @@ def canonicalize_dataset_by_cli(
                     )
             self._print_divider()
 
-        dataset = datasets.load_dataset(dataset_name, chosen_config)
+        dataset = datasets.load_dataset(dataset_name, chosen_config).flatten()
+
         if "train" not in dataset:
             raise ValueError("The dataset must contain a `train` split.")
+
+        columns_mapping = {
+            col: col.replace(".", "_") for col in dataset["train"].column_names
```

Comment:
What will happen if there is a column conflict here? E.g. we have two columns, `foo.baz` and `foo_baz`? Then I think these will conflict and one of the columns will be erased.

I know this is unlikely but might be good to check and handle this scenario.